### DISCUSSION TITLE
System Integration Elasticsearch Index Template Typo

### DISCUSSION THREAD

**Discussion by EddieN17** [Security-Onion-Solutions/securityonion#11125] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### Network Traffic Collection

tap

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

I had an issue with the Elastic Agent system integration, where if the data stream is rolled over, the integration just stops ingesting those logs. I believe I found the issue, at least it fixed the problem for me, by updating `salt/elasticsearch/defaults.yaml` as some of the index patterns are missing a `-` towards the end of them. My guess is the logs weren't properly getting caught by the index templates.

So I added the ending character to:
`logs-system.auth-*`
`logs-system.syslog-*`
`logs-system.system-*`
`logs-system.application-*`
`logs-system.security-*`
`logs-windows.forwarded-*`

These are also how the default elasticsearch index patterns are typed.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by TOoSmOotH** [comment]
Did you modify the file directly or just change it in the UI?

**Reply by EddieN17** [reply]
I made the edits in the UI, I was just referring to the `salt/elasticsearch/defaults.yaml` as the direct file to change for future versions.

**Reply by weslambert** [reply]
I'm not sure why the character change would affect anything with rollovers, and it should still match without the hyphen. 

If it was able to ingest and events were processed through the proper pipelines initially, then I wouldn't think there would be an issue with the pattern. It feels like maybe there is something else is happening. 

Did you witness any error messages within logs to point you in that direction?

**Reply by EddieN17** [reply]
So, my security integration stopped showing any logs at the relative moment my data stream rolled over, so I was fairly confident those two are linked. 
Here is the error on an endpoint
```
{"log.level":"error","@timestamp":"2023-08-21T12:14:38.824-0500","message":"Failed to publish events caused by: write tcp (ip):47380->(ip):5055: wsasend: An existing connection was forcibly closed by the remote host.","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"winlog-default","type":"winlog"},"log":{"source":"winlog-default"},"service.name":"filebeat","ecs.version":"1.6.0","log.logger":"logstash","log.origin":{"file.line":280,"file.name":"logstash/async.go"},"ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2023-08-21T12:14:40.567-0500","message":"failed to publish events: write tcp (ip):47380->(ip):5055: wsasend: An existing connection was forcibly closed by the remote host.","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"winlog-default","type":"winlog"},"log":{"source":"winlog-default"},"log.origin":{"file.line":174,"file.name":"pipeline/client_worker.go"},"service.name":"filebeat","ecs.version":"1.6.0","log.logger":"publisher_pipeline_output","ecs.version":"1.6.0"}
```

I rebooted my systems and the integration was still showing no logs. Now this is where I lose all factual evidence, but a year ago, I remember playing with elasticsearch's wildcard, and it was extremely picky with the '-' character. If I ended a wildcard search with the * character but the next character ended up being '-' it wouldn't include it in the filter. Now there's no evidence that shows that still is the problem, it's just something I remembered. 

But if the index pattern doesn't have the '-' and you manually `POST logs-system.security-default/_rollover` in the kibana dev console, the new index doesn't get any populated logs. You can reinstall the system integration in kibana, and then it'll work for the one data stream again, but if it rolls over again, it's going to not populate the data again. But if you do add the '-' to the index pattern, manually rollover the data steam again and reinstall the system integration in kibana, logs get populated after every rollover you do. This is all on a hunch I had a year ago, and somehow the results are still correlating with it. I know my error log doesn't really correlate with this, but I'm guessing the manager just force closes the connection after it realizes it doesn't have a data stream that matches the logs its trying to send.

I couldn't get around without reinstalling the integration either way, so maybe there's some variable there that just seems to cure my results.

**Reply by weslambert** [reply, marked as answer]
Part of the issue you are experiencing may be related to a fix that is included with the a hotfix release. 

https://blog.securityonion.net/

It is likely that  it was required to reinstall the integration because of a mismatch in the integration version and the Elasticsearch ingest node pipelines. When the integration is reinstalled, a `force` flag is passed to the API endpoint, as opposed to the initial version package upgrade script. 

The hotfix ensures that the versions are in alignment by modifying the package upgrade script to force installation of the package.

I also just tried a manual rollover of `logs-system.syslog`, which does not include a hyphen in the index pattern, and it rolled over fine and is continuing to index data.

**Reply by EddieN17** [reply]
Sounds good, I'll give it a try. Thanks!
